### PR TITLE
Added the ability for the /prices endpoint to include details about the promotions which determine the returned prices

### PR DESCRIPTION
**Pull request by paulbrown1982**
## What are you doing in this PR?

Added the ability for the `/prices` endpoint endpoint to include, when the query parameter `?include-summary` is provided, details about the promotions which determine the returned prices.

Also added currency as a price should never be referenced without its currency being alongside.

## Why are you doing this?
This is to enable the currently displayed offers on the Digital Subscription and Guardian Weekly checkouts to be shown elsewhere, for example a cancellation-save journey.

## Is this an AB test?
- [ ] Yes
- [x] No

If this is an AB test, PR reviewers should open and check the Optimize test.
[**Optimize Link**](https://optimize.google.com/optimize/home)

## Accessibility test checklist
 - [ ] [Tested with screen reader](https://accessibility.gutools.co.uk/testing/web/screen-readers/)
 - [ ] [Navigable with keyboard](https://accessibility.gutools.co.uk/testing/web/keyboard-navigation/)
 - [ ] [Colour contrast passed](https://accessibility.gutools.co.uk/testing/web/colour-contrast/)

## Screenshots
N/A

### CODE REVIEW

File: support-frontend/app/controllers/PricesController.scala [hunk 1]
```diff
@@ -47,19 +51,26 @@ object PricesController {
...
+    val price = topPromotion
       .flatMap(_.discountedPrice)
       .getOrElse(priceSummary.price)
 
-    RatePlanPriceData(price.toString)
+    RatePlanPriceData(
+      price = price.toString,
+      currency = priceSummary.currency.identifier,
+      priceSummary =
+        Option.when(includeSummary)(priceSummary.copy(promotions = topPromotion.map(_.copy(landingPage = None)).toList)),
```
Comment (tomrf1) [inline support-frontend/app/controllers/PricesController.scala:64]: `Option.when` is new to me!

File: support-frontend/conf/routes [hunk 2]
```diff
@@ -170,6 +170,7 @@ GET /p/:promoCode                                                 controllers.Pr
 
 # ----- Prices API ----- #
 GET /prices                                                       controllers.PricesController.getPrices()
+GET /pricesWithSummary                                            controllers.PricesController.getPricesWithSummary()
```
Comment (tomrf1) [inline support-frontend/conf/routes:173]: since it's almost the same it could even be `/prices?show-summary`

Comment (paulbrown1982) [reply support-frontend/conf/routes:173]: That was my original plan, however upon trying I assumed it wasn't possible/advisable to read query parameters with a `CachedAction`, and I didn't want to change it to a `NoCacheAction`?

Comment (tomrf1) [reply support-frontend/conf/routes:173]: Ah right. Perhaps we should change this to a regular `Action`? Currently [SDC polls](https://github.com/guardian/support-dotcom-components/blob/main/packages/server/src/productPrices.ts#L15) the endpoint every minute (for each instance) which isn't much. I guess it depends what it's going to be used for long term

Comment (paulbrown1982) [reply support-frontend/conf/routes:173]: Ok I'll do that.

**Review by tomrf1 — APPROVED** [review]
looks good 👍

**Pull request merged**